msm-sitemap.php

```diff
@@ -88,6 +88,14 @@ public static function sitemap_rewrite_init() {
 		} else {
 			add_rewrite_rule( '^sitemap.xml$','index.php?sitemap=true','top' );
 		}
+
+		/**
+		 * Intended for any additional rewrite rules that might be applied for extensibilty
+		 * The intention is to call add_rewrite_rules on more sitemap types
+		 * ie news-sitemap.xml etc etc
+		 */
+		do_action('msm_additional_sitemap_rewrites');
```

Comment:
Renamed this as per your suggestion, we can look into investing the time for the array functionality you have mentioned, I agree that it would be a more robust solution